### ISSUE TITLE
Use exact matching instead of pattern matching

### ISSUE THREAD

**Issue by nassibnassar** [library-data-platform/ui-ldp#33]
When this app was initially written, we used the SQL `ILIKE` operator and `%` wildcard for query filtering, in order to support case-insensitive substring matches.  In LDP's current, pre-Metadb code base, it appears that we cannot assume that columns will be indexed for pattern matching, due to the lack of incremental updates, and as a result we will have to support exact matching only.

The relevant code appears to be in `mod-ldp/src/main/java/org/folio/ldp/QueryController.java`:
```
if(availableColumns.get(col.key).equals("character varying")) {
  selectQuery = selectQuery.addCondition(PgBinaryCondition.iLike(new CustomSql(col.key), "%"+col.value+"%"));
} else {
  selectQuery = selectQuery.addCondition(BinaryCondition.equalTo(new CustomSql(col.key), col.value));
}
```

The change may be as simple as replacing the above code with:
```
selectQuery = selectQuery.addCondition(BinaryCondition.equalTo(new CustomSql(col.key), col.value));
```

In the future, with Metadb, it should be possible to support pattern matching because the indexes can be updated incrementally.  So an alternative to the above change might be to enable pattern matching only when using Metadb, e.g. changing the first line to:
```
if (METADB && availableColumns.get(col.key).equals("character varying")) {
```

At some point this would have to be coordinated with the UI in some way.  Maybe the operator, `=` vs. `~`(?), could be selectable in a dropdown list or some other, more user-friendly way.  If running with the current LDP code base, `=` would be the only available choice.  If running with Metadb, both may be available (depending on Metadb configuration); that could be queryable in the Metadb protocol later.

**Comment by MikeTaylor** [comment]
Blocked on https://github.com/library-data-platform/mod-ldp/issues/13 since this is really a back-end issue.

**Comment by MikeTaylor** [comment]
Closing since the back-end issue (where this actually needed fixing) is closed.

The stuff about maybe one day offering the user a choice of `=` and `~` when MetaDB is in use — that can become its own separate issue when that day comes.
